packages/workspace/src/browser/workspace-service.ts

```diff
@@ -21,21 +21,30 @@ import { FileSystemWatcher } from '@theia/filesystem/lib/browser';
 import { WorkspaceServer } from '../common';
 import { WindowService } from '@theia/core/lib/browser/window/window-service';
 import { FrontendApplication, FrontendApplicationContribution } from '@theia/core/lib/browser';
+import { Disposable, Emitter, Event } from '@theia/core/lib/common';
 import { Deferred } from '@theia/core/lib/common/promise-util';
 import { ILogger } from '@theia/core/lib/common/logger';
 import { WorkspacePreferences } from './workspace-preferences';
 
+export interface WorkspaceDescription {
+    root: FileStat | undefined;
+    roots: FileStat[];
+    workspaceConfigFile: FileStat | undefined;
+}
+
 /**
  * The workspace service.
  */
 @injectable()
 export class WorkspaceService implements FrontendApplicationContribution {
 
     private _root: FileStat | undefined;
+    private _workspaceDescription: WorkspaceDescription;
 
-    private readonly deferredRoot = new Deferred<FileStat | undefined>();
+    private deferredRoot = new Deferred<FileStat | undefined>();
```

Comment:
These should become the `deferredRoots`.

Reply:
it is nice if we only need to expose the `roots` and nothing else. however the `workspaceService.roots` right now is counting on the readiness of `preferences`, while `preferences` needs `workspaceService` to tell it where to retrieve the settings - it is a circular dependency. 

if you dislike the idea of having `workspace.supportMultiRootWorkpsace` to toggle the multi-root support, we could implement the toggle by adding a command line argument. That way the user would need to run 
```
yarn start --multi-root
```
to enable the multi-root support, instead of changing the preference. 

What do you think? @svenefftinge @akosyakov